WooCommerce/Classes/ViewRelated/Orders/Order Details/Shipping Labels/WooShipping Create Shipping Labels/WooShipping Package and Rate Selection/WooShippingAddPackageView.swift

```diff
@@ -64,100 +92,199 @@ struct WooShippingAddPackageView: View {
-                        self.fieldValues[dimensionType] = value
-                    }))
+                .onChange(of: packageTemplateNameFieldFocused) { focused in
+                    if focused {
+                        DispatchQueue.main.asyncAfter(deadline: .now() + Constants.scrollToDelay, execute: {
+                            withAnimation {
+                                proxy.scrollTo(Constants.saveTemplateButtonID, anchor: .bottom)
+                            }
+                        })
+                    }
                 }
             }
-            Toggle(isOn: $showSaveTemplate) {
-                Text(Localization.saveNewPackageTemplate)
-                    .font(.subheadline)
-            }
-            if showSaveTemplate {
-                Button(Localization.savePackageTemplate) {
-                    // save template
-                }
-                .buttonStyle(SecondaryButtonStyle())
+        }
+        if !showSaveTemplate {
+            Spacer()
+            Button(Localization.addPackage) {
+                addPackageButtonTapped()
             }
+            .disabled(!validateCustomPackageInputFields())
+            .buttonStyle(PrimaryButtonStyle())
+            .padding()
+        }
+    }
+
+    private func onBackwardButtonTapped() {
+        switch focusedField {
+        case .length:
+            return
+        case .width:
+            focusedField = .length
+        case .height:
+            focusedField = .width
+        case nil:
+            return
+        }
+    }
+
+    private func onForwardButtonTapped() {
+        switch focusedField {
+        case .length:
+            focusedField = .width
+        case .width:
+            focusedField = .height
+        case .height:
+            return
+        case nil:
+            return
         }
-        .padding(.horizontal)
     }
 
+    private func dismissKeyboard() {
+        focusedField = nil
+        packageTemplateNameFieldFocused = false
+    }
+
+    @ViewBuilder
     private var carrierPackageView: some View {
         // TODO: just a placeholder
-        Text("carrier")
+        Spacer()
     }
 
+    @ViewBuilder
     private var savedPackageView: some View {
         // TODO: just a placeholder
-        Text("saved")
+        Spacer()
+    }
+
+    // MARK: - actions
+
+    private func validateCustomPackageInputFields() -> Bool {
+        guard !customPackageViewModel.areFieldValuesInvalid else {
+            return false
+        }
+        if showSaveTemplate {
+            return !packageTemplateName.isEmpty
+        }
+        return true
+    }
+
+    private func addPackageButtonTapped() {
+        // TODO: implement adding a package
+        guard validateCustomPackageInputFields() else { return }
+    }
+
+    private func savePackageAsTemplateButtonTapped() {
+        // TODO: implement saving package as a template
+        guard validateCustomPackageInputFields() else { return }
```

Comment:
I'd suggest putting actions like these in the view model, so we can unit test them.